### ISSUE TITLE
TypeScript error om importing redux-thunk

### ISSUE THREAD

**Issue by born2net** [reduxjs/redux-thunk#75]
with latest version of redux-thunk I am getting TypeScript error of:

```
Error:(24, 61) TS2304: Cannot find name 'thunkMiddleware'.
```

I already installed typings redux-thunk and included it.
here is the config:

![capture](https://cloud.githubusercontent.com/assets/5090936/15628839/e2f34594-24c0-11e6-95e9-adc8d7a661df.PNG)

so it's not seeing the imported member,

as workaround added to definition file:

```
declare var thunkMiddleware:any;
```

regards

Sean

**Comment by garthk** [comment]
Full error:
```
src/client/reducers.ts(20,53): error TS2345: Argument of type 'Thunk' is not assignable to parameter of type 'Middleware'.
  Type 'Thunk' provides no match for the signature '<S>(api: MiddlewareAPI<S>): (next: Dispatch<S>) => Dispatch<S>'
typings/globals/redux-thunk/index.d.ts(4,36): error TS2503: Cannot find namespace 'Redux'.
typings/globals/redux-thunk/index.d.ts(6,21): error TS2503: Cannot find namespace 'Redux'.
```

Manual typings to fix the problem:
```
declare module 'redux-thunk' {
    import { Middleware } from 'redux';
    const thunkMiddleware : Middleware;
    export default thunkMiddleware;
}
```

I tried matching the namespace from `npm~react-thunk`, but ran into `invalid module name in augmentation` until my head bled.

**Comment by brauliodiez** [comment]
Thanks @garthk for the temporary fix, from my side I have removed the typings entry from redux and using the one that redux is providing when you make an npm install. I have noticed redux-thunk in GitHub contains and index.d.ts but it doesn't get downloaded on npm install, is something that will be added to the package? Do we have to wait for a typings update? I have tried to paste that .d.ts manually and tweak it a bit but I don't manage to get it working properly.

**Comment by lrojas94** [comment]
I used the DefinitelyTyped typings for redux-thunk and had no issues whatsoever. I believe you can install it using typings by writing ``` typings install -SG dt~redux-thunk ```. You can read more about it here:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/redux-thunk/redux-thunk.d.ts

**Comment by brauliodiez** [comment]
Hi @lrojas94 which version of redux are you targeting? (in my case 3.5.2 and using redux definitions directly from the npm). It worked for me as well but on previous versions.

**Comment by lrojas94** [comment]
@brauliodiez  I'm targeting the same, 3.5.2, but I'm using DefinitelyTyped typings.

**Comment by brauliodiez** [comment]
Mmmh, I think if a project itself offers a d.ts maybe we should follow that definition instead of the typings ones, reasons for that: the project definition should be uptodate with the version cut that you are using, the typings one maybe not (this can apply for cases like Redux that is a very well known library).

You can follow the Redux typescript PR here:

https://github.com/gaearon/redux-thunk/pull/77

What you do think? Makes sense? Or am I missing something?

**Comment by lrojas94** [comment]
Sidenote: I will be referring to Definitely Typed as DT

I believe it would make sense if the Redux-Thunk typings are using a standard typings from Redux, however, I believe that's not the way it works as of now. The **Redux** definitions themselves are from DT (at least the ones I'm using), so, aren't the Redux-Thunk typings from DT more likely to be compatible with the **Redux** typings?

**Comment by brauliodiez** [comment]
That's the strange thing, the one from redux github site is (this seems to break redux-thunk amongst others):

https://github.com/reactjs/redux/blob/master/index.d.ts

The one from typigns:

https://github.com/rangle/typings-redux/blob/master/redux.d.ts

I think I'm missing something here that I don't understand.

**Comment by lrojas94** [comment]
This is the one I'm currently using:

**Comment by unional** [comment]
It is because a new release is needed to get the typings into the package.

For now you can do `typings install -S github:gaearon/redux-thunk/index.d.ts` to get around it until then.

**Comment by noxan** [comment]
Is there something holding back a release containing the type definitions? Just asking since I would like to get rid of manually installing the type definitions in addition to the package.

**Comment by aikoven** [comment]
Released [v2.1.1](https://github.com/gaearon/redux-thunk/releases/tag/v2.1.1) with TypeScript definitions included.
